### PR TITLE
Add ClickHouse v22.8

### PR DESCRIPTION
**Pull request by rschu1ze**

### CODE REVIEW

File: Formula/clickhouse@22.8.rb [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  head "https://github.com/ClickHouse/ClickHouse.git",
+    branch:   "22.8"
+
+  livecheck do
+    url :stable
+    regex(/^v?(22\.7(?:\.\d+)+)-(?:stable|lts)$/i)
+  end
+
+  bottle do
+    root_url "https://github.com/Altinity/homebrew-clickhouse/releases/download/clickhouse@22.7-22.7.2.15"
```
Comment (alexey-milovidov) [inline Formula/clickhouse@22.8.rb:17]: This looks incorrect.

Comment (rschu1ze) [reply Formula/clickhouse@22.8.rb:17]: Yepp, thanks.

**Comment by rschu1ze** [comment]
Currently getting

```
[...]
-- Performing Test SUPPORTS_CFLAG_no_enum_constexpr_conversion
-- Performing Test SUPPORTS_CFLAG_no_enum_constexpr_conversion - Success
-- No official build: A checksum hash will not be added to the clickhouse executable
-- Default libraries: -nodefaultlibs  -lc -lm -lpthread -ldl
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - no
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - not found
CMake Error at /opt/homebrew/Cellar/cmake/3.26.4/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
Could NOT find Threads (missing: Threads_FOUND)
Call Stack (most recent call first):
/opt/homebrew/Cellar/cmake/3.26.4/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
/opt/homebrew/Cellar/cmake/3.26.4/share/cmake/Modules/FindThreads.cmake:226 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
cmake/darwin/default_libs.cmake:23 (find_package)
CMakeLists.txt:440 (include)

-- Configuring incomplete, errors occurred!
```

**Comment by rschu1ze** [comment]
Happens also when brewing the existing v22.7 :-(

**Comment by rschu1ze** [comment]
Steps:

```
brew tap clickhouse/clickhouse
cd $(brew --repository clickhouse/clickhouse)
export HOMEBREW_NO_INSTALL_FROM_API=1
brew install --verbose --build-from-source clickhouse/clickhouse/clickhouse@22.7
```

**Comment by alexey-milovidov** [comment]
Is it possible to ignore everything Homebrew related and build ClickHouse as close as possible to the official instruction?

**Comment by rschu1ze** [comment]
It fails also with vanilla build instructions 😢 

@traceon Don't want to bother you too much, it looks like you moved on to another job. The homebrew repo was recently moved from Altinity to ClickHouse. My [attempt](https://github.com/ClickHouse/homebrew-clickhouse/pull/59#issuecomment-1601761152) to compile the last included version v22.7 failed (see above). But since I am not great with Homebrew, I possibly missed something. Maybe you remember the steps to build the code locally, i.e. without GitHub runners?

**Comment by rschu1ze** [comment]
@traceon provided a viable workaround (thanks) for the pthread issue.

I now get lots of compiler errors when building v22.8 with clang-16. I guess we should skip the old versions and jump directly to v22.5. Will do that in a new PR.

**Pull request closed without merging**